pkg/router/template/router.go

```diff
@@ -171,13 +172,23 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 }
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	// IDhash contains an obfuscated internal IP address that is the value
+	// passed in the cookie. The IP address is made more difficult to extract
+	// by including other internal information in the hash.
 	if len(alias.PreferPort) == 0 {
+		for i := range svc.EndpointTable {
+			endpoint := svc.EndpointTable[i]
```

Comment:
I think you missed @eparis' comment about the go idioms.

It's ugly that we have to do the same thing in two places though.  Perhaps make a private function that computes a hash value from an Endpoints structure?